examples/MerminMagic/main.jl

```diff
@@ -0,0 +1,59 @@
+# Mermin-Peres Magic Square Game Algorithm
+# https://www.gregegan.net/SCIENCE/MPMS/MPMS.html
+using Yao
+using Yao.EasyBuild, YaoPlots
+using Random
+
+# initialize four qubits to be two pairs of Bell Pairs (00)
+function init_double_bell_state(reg::ArrayReg)
+    for k in [1, 3]
+        apply!(reg, put(4, k => H))
+        apply!(reg, cnot(4, k, k + 1))
+    end
+end
+
+function meas_by_idx(reg::ArrayReg, i::Int, j::Int)
+    # measurement operator table
+    op_mtx = Matrix{KronBlock}(undef, 3, 3)
+    op_mtx = [kron(I2, Z) kron(Z, I2) kron(Z, Z); kron(X, I2) kron(I2, X) kron(X, X); kron(-X, Z) kron(-Z, X) kron(Y, Y)]
+
+    ans = zeros(Float64, 4)
+    for k in 1:2
+        # do measurement for Alice depending on the index given
+        ans[k] = measure!(op_mtx[i, k], reg, (1, 3))[1]
+        # do measurement for Bob depending on the index given
+        ans[k+2] = measure!(op_mtx[k, j], reg, (2, 4))[1]
+    end
+
+    return real.(ans)
```

Comment:
real is not necessary here.

Reply:
If I don't have this line, when I output a1 a2 b1 b2, I will get a complex number that seems quite ugly to me.